File: newFive/src/main/java/formation/projetParis/newFive/entities/Ballon.java
package formation.projetParis.newFive.entities;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@DiscriminatorValue("B")
public class Ballon extends Marchandise {

	@JsonView(JsonViews.Base.class)
    private String couleur;
	@JsonView(JsonViews.Base.class)
    private String taille;
    
    public Ballon() {
    }
    
	public Ballon(String couleur, String taille) {
		super();
		this.couleur = couleur;
		this.taille = taille;
	}

	public String getCouleur() {
		return couleur;
	}

	public void setCouleur(String couleur) {
		this.couleur = couleur;
	}

	public String getTaille() {
		return taille;
	}

	public void setTaille(String taille) {
		this.taille = taille;
	}
    
	
    
    

}

File: newFiveAngular/src/app/model/ballon.ts
import { Marchandise } from "./marchandise";

export class Ballon extends Marchandise{
    constructor(
      private _taille?: string,
      private _couleur?: string
    ) { super()
    }
  
    public get taille(): string | undefined {
      return this._taille;
    }
    public set taille(value: string | undefined) {
      this._taille = value;
    }
  
    public get couleur(): string | undefined {
      return this._couleur;
    }
    public set couleur(value: string | undefined) {
      this._couleur = value;
    }
  }
  